src/javascript/detectors/sendfile_injection/sendfile_injection.js

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// {fact rule=sendfile-injection@v1.0 defects=1}
+var express = require("express")
+var path = require("path")
+var app = express()
+function sendfileInjectionNoncompliant() {
+    app.get('www.example.com', (req, res) => {
+        var fileName = req.params.file
+        // Noncompliant: tainted-data is passed into 'res.sendfile'.
+        res.sendFile(fileName)
+    })
+}
+// {/fact}
+
+
+// {fact rule=sendfile-injection@v1.0 defects=0}
+var express = require("express")
+var path = require("path")
+var app = express()
+function sendfileInjectionCompliant() {
+    app.get('www.example.com', (req, res) => {
+        var fileName = "file.txt"
+        if (fileName !== req.params.file) {
+            console.log("Valid file name")
+        } else {
+            console.log("Invalid file name")
+        }
+        // Compliant: validated fileName before passing into 'res.sendFile'.
+        res.sendFile(fileName)
```

Comment:
Can we put this line under `if (fileName !== req.params.file) {` since we are validating the file and then passing it to response? Also, we can throw error in the else part instead of only logging.